### PR TITLE
test(integration): angular 5.2 + typescript 2.7

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

https://github.com/ngxs/store/issues/847

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by markwhitfeld** [comment]
Just a note that, as discussed, Angular 5.2 does officially not support TS 2.7,  but there is an existing issue #357 that has been around for a while that points out why we upgraded to TS2.7 as a workaround (at least for now).
Also, upgrading rxjs with the compat lib fixed issue #270 found in this integration app.

**Comment by markwhitfeld** [comment]
@splincode I recommend just adding a 'pre*' npm script step for the specific integration build task that copies the `@ngxs` folder into the `node_modules` folder of the ng 5 integration app.
This worked for me and allowed me to identify and fix those issues.

**Comment by splincode** [comment]
We can't support TypeScript 2.7, because my build does not work at all for the new version of NGXS (3.4.0-dev): https://github.com/ReactiveX/rxjs/issues/4512

I propose to make an announcement where we indicate the versions that we can minimally support.

Minimal TypeScript - 2.8
Minimal Angular - 5.2
Minimal RxJS - 6.2, or (rxjs - 5.6.0-forward-compat.4, rxjs-compat - 6.2.2)

**Comment by splincode** [comment]
integration test is ready

File: packages/hmr-plugin/src/public_api.ts [hunk 1]
```diff
@@ -1,4 +1,10 @@
-export { NgxsHmrLifeCycle, NgxsHmrOptions, NgxsHmrSnapshot } from './symbols';
 export { HmrInitAction } from './actions/hmr-init.action';
+export {
+  NgxsHmrLifeCycle,
+  NgxsHmrOptions,
+  NgxsHmrSnapshot,
+  WebpackModule,
+  BootstrapModuleFn
```
Comment (markwhitfeld) [inline packages/hmr-plugin/src/public_api.ts]: Why have `WebpackModule` and `BootstrapModuleFn` been added to the exports?
Was something not working? It works fine in my app without this.

File: integrations/hello-world-ng5/package.json [hunk 2]
```diff
@@ -50,6 +42,6 @@
     "protractor": "5.1.2",
     "ts-node": "4.1.0",
     "tslint": "5.9.1",
-    "typescript": "2.7.1"
+    "typescript": "2.8.1"
```
Comment (markwhitfeld) [inline integrations/hello-world-ng5/package.json]: What was the reason for this version bump?
I had it working with 2.7.1

Comment (splincode) [reply integrations/hello-world-ng5/package.json]: https://github.com/ReactiveX/rxjs/issues/4512#issuecomment-464892499

I did the same thing, copied the directories, it did not work, I did it through the paths, but it did not work there

File: packages/hmr-plugin/src/public_api.ts (hunk 1 above)
Comment (splincode) [reply packages/hmr-plugin/src/public_api.ts]: no, it works without, I just wanted to use the types in the example, the user may also want this

**Comment by splincode** [comment]
@markwhitfeld ping

**Comment by markwhitfeld** [comment]
Giving an error when I try to start this integration app:
image

**Comment by splincode** [comment]
@markwhitfeld we need fix it

**Comment by splincode** [comment]
I think we need to add unit tests for this integration test.

**Comment by markwhitfeld** [comment]
Ok, I got the router working. Fixed the issue above.
@splincode I agree about some unit tests. Could you add some. I think just a basic AppComponent test that dispatches an action and then checks that it is rendered on the component would suffice.

**Comment by markwhitfeld** [comment]
One last issue I have picked up...
App starts and serves great, but when I make a change and HMR pushed the update then I get this error:
image

@splincode could you try to reproduce and see if there is a bug in HMR? ... maybe HMR doesn't support Angular 5?

**Comment by markwhitfeld** [comment]
@splincode will fix the hmr issue with NG5 in a separate PR.

**Pull request merged**